tests/cms/models/events/test_recurrence_rule.py

```diff
@@ -0,0 +1,108 @@
+"""
+Test module for RecurrenceRule class
+"""
+import datetime
+
+import pytz
+
+from integreat_cms.cms.models import RecurrenceRule, Event
+
+
+# pylint: disable=missing-docstring
+
+
+class TestCreatingIcalRule:
+    """
+    Test whether to_ical_rrule_string function is calculating the rrule corretly
+    """
+
+    test_event = Event(
+        start=datetime.datetime(2030, 1, 1, 11, 30, 0, 0, pytz.UTC),
+        end=datetime.datetime(2030, 1, 1, 12, 30, 0, 0, pytz.UTC),
+    )
+
+    def test_api_rrule_every_year_start_date_in_the_past(self):
```

Comment:
> > There is just one edge case when the initial occurrence of an recurrent event is already in the past, then we would completely hide this event in the app if we omit the future occurrences in the API.
>
> Fix this and added a test for this case.

Actually, this test case doesn't really cover the previous bug (since the real API code is not executed here)... if we really wanted to make sure this isn't reintroduced in the future, we would have to add a past event to the test data and adapt all the expected API responses again... I'll leave this up to you if you're still motivated :sweat_smile:

Reply:
You are right, actually this only shows me, there is too much logic in the endpoints, that should be in the models.

Reply:
Yes, agreed :grin: